bin/ossh

```diff
@@ -0,0 +1,244 @@
+                          action="store_true", help="Verbose?")
+        parser.add_argument('--list', default=False,
+                          action="store_true", help="list out hosts")
+        parser.add_argument('-c', '--command', action='store',
+                            help='Command to run on remote host')
+        parser.add_argument('-l', '--login_name', action='store',
+                            help='User in which to ssh as')
+
+        parser.add_argument('-o', '--ssh_opts', action='store',
+                            help='options to pass to SSH.\n \
+                                  "-o ForwardX11 yes"')
+        parser.add_argument('host', nargs='?', default='')
+
+        self.args = parser.parse_args()
+        self.parser = parser
+
+
+    def process_host(self):
+        '''Determine host name and user name for SSH.
+        '''
+        self.env = None
+        self.user = None
+
+        re_env = re.compile('\.(int|stg|prod|ops)')
+        search = re_env.search(self.args.host)
+        if self.args.env:
+            self.env = self.args.env
+        elif search:
+            # take the first?
+            self.env = search.groups()[0]
+
+        # remove env from hostname command line arg if found
+        if search:
+            self.args.host = re_env.split(self.args.host)[0]
+
+        # parse username if passed
+        if '@' in self.args.host:
+            self.user, self.host = self.args.host.split('@')
+        else:
+            self.host = self.args.host
+            if self.args.login_name:
+                self.user = self.args.login_name
+
+    def get_hosts(self, cache=False):
+        '''Query our host inventory and return a dict where the format
+           equals:
+
+           dict['servername'] = dns_name
+        '''
+        # TODO: perform a numerical sort on these hosts
+        # and display them
+
+        if not cache:
+            self.host_inventory = self.ansible.build_host_dict()
+        else:
+            self.host_inventory = self.ansible.build_host_dict(['--cache-only'])
+
+    def select_host(self, regex=False):
+        '''select host attempts to match the host specified
+           on the command line with a list of hosts.
+
+           if regex is specified then we will attempt to match
+           all *{host_string}* equivalents.
+        '''
+# list environment stuff as well
+# 3 states:
+#  - an exact match; return result
+#  - a partial match; return all regex results
+#  - no match; None
+        re_host = re.compile(self.host)
+
+        exact = []
+        results = []
+        for hostname, server_info in self.host_inventory[self.env].items():
+            if hostname.split(':')[0] == self.host:
+                exact.append((hostname, server_info))
+                break
+            elif re_host.search(hostname):
+                results.append((hostname, server_info))
+
+        if exact:
+            return exact
```

Comment:
I prefer to not use a variable for exact here and just return from the for loop conditional above. Just a preference.